example/game/src/main/java/no/nordicsemi/android/ble/example/game/server/viewmodel/ServerViewModel.kt

```diff
@@ -0,0 +1,234 @@
+
+            override fun onDeviceConnectedToServer(device: BluetoothDevice) {
+                ServerConnection(context, viewModelScope, device)
+                    .apply {
+                        stateAsFlow()
+                            .onEach { connectionState ->
+                                val currentState = _state.value
+                                when (connectionState) {
+                                    ConnectionState.Ready -> {
+                                        clients.value += this
+                                        _state.value = WaitingForPlayers(clients.value.size)
+                                    }
+                                    is ConnectionState.Disconnected -> {
+                                        clients.value -= this
+
+                                        when (currentState) {
+                                            is WaitingForPlayers -> {
+                                                _state.value = WaitingForPlayers(clients.value.size)
+                                            }
+                                            else -> {
+                                                // TODO handle disconnection during game
+                                            }
+                                        }
+                                    }
+                                    else -> {}
+                                }
+                            }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        playersName
+                            .onEach { savePlayersName(it) }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        result
+                            .onEach { saveScore(it) }
+                            .launchIn(viewModelScope)
+                    }
+                    .apply {
+                        useServer(serverManager)
+
+                        val exceptionHandler = CoroutineExceptionHandler { _, throwable ->
+                            Log.e(TAG, "Error", throwable)
+                        }
+                        viewModelScope.launch(Dispatchers.IO + exceptionHandler) {
+                            connect()
+                        }
+                    }
+
+            }
+
+            override fun onDeviceDisconnectedFromServer(device: BluetoothDevice) {
+                Log.w("Device Disconnected", "Device Disconnected From the Server!")
+            }
+
+        })
+        serverManager.open()
+    }
+
+    fun savePlayersName(playerName: String) {
+        if (playerName != "") {
+            savedResult.value += FinalResult(
+                name = playerName,
+                score = 0
+            )
+            viewModelScope.launch {
+                clients.value.forEach {
+                    it.gameOver(ResultToClient(false, savedResult.value))
```

Comment:
What does it do?
You.. send "Game over" with "isGameOver == false"? and some result?
Shouldn't the method just save player's name?